### PR TITLE
Weather weather weather weather

### PR DESCRIPTION
**Pull request by marusak**
https://mmarusak.fedorapeople.org/tests.html (with the current DB)

### CODE REVIEW

File: store-tests [hunk 1]
```diff
@@ -82,7 +82,7 @@ def main():
     # 'repo', 'url', 'state', 'description'
     # 'started' - when this run was picked by bot
     # 'posted' - when this run was triggered
-    # 'finished' - when this run finished - 'state' is then either 'success', 'failure' or 'error'
+    # 'finished' - when this run finished - 'state' is then either 'pending', 'success', 'failure' or 'error'
```
Comment (martinpitt) [inline store-tests:85]: That comment is a bit confusing -- The "finished" field does not exist for state "pending"; move the comment to "posted"?

File: store-tests [hunk 2]
```diff
@@ -117,36 +118,39 @@ def main():
...
+            waited = started - posted
+            took = status.get("finished", started) - started
             logging.debug("Processing {0}{1} as {2} that waited {3}s and took {4}s in {5}. {6}".format(
                 context,
                 "(retry no." + str(retry) + ")" if retry else "",
                 status["state"] + (" (" + status["description"] + ")" if status.get("description") else ""),
                 waited, took, repo, url
             ))
 
+            # URL is our primary key`, so if we don't have one, just fake something unique
```
Comment (martinpitt) [inline store-tests:133]: stray `

File: store-tests [hunk 3]
```diff
@@ -117,36 +118,39 @@ def main():
...
+            if not url:
+                url = "-".join(["xxx", opts.revision, context, str(retry)])
```
Comment (martinpitt) [inline store-tests:135]: xxx → queued or pending?

File: store-tests [hunk 4]
```diff
@@ -117,36 +118,39 @@ def main():
...
                 continue
 
             run_id = insert_run(cursor, repo, opts.revision, context, url,
-                                status["posted"], retry, waited, took, status["state"],
+                                posted, retry, waited, took, status["state"],
                                 status.get("description"))
 
-            raw_log = url[:-5] if url.endswith(".html") else url
-            logging.debug("scanning log %s", raw_log)
+            if (status.get("url")):
```
Comment (martinpitt) [inline store-tests:146]: My first instinct here was "this should be `if url:`, but of course we don't want to scan fake URLs. A small comment may be helpful for our future selves. Or better yet, move this up as `else:` to the place that sets the fake url., and then conditionalize this on `raw_url`.

BTW, this is python, no need to wrap the condition into ().

File: tests.html [hunk 5]
```diff
@@ -276,7 +276,7 @@ <h2 id="g4-title"></h2>
                             SELECT testname, COUNT(run) as failed
                             FROM Tests
                             JOIN TestRuns ON Tests.run = TestRuns.id
-                            WHERE TestRuns.time > ${since} AND Tests.failed = 1 AND project LIKE '${repo}'
+                            WHERE TestRuns.time > ${since} AND Tests.failed = 1 AND project LIKE '${repo}' AND testname NOT LIKE '%selenium%'
```
Comment (martinpitt) [inline tests.html:279]: Can you please add a commit message (or comment) like "The selenium tests have a bug where skipped tests show up as failures. They are deprecated and not very flaky, so not interesting to track."

**Review by martinpitt — CHANGES_REQUESTED** [review]
Superb, dakujem! Looks a lot nicer :grin:

File: store-tests (hunk 4 above)
Comment (marusak) [reply store-tests:146]: Right, I want to keep also checks for already existing, mostly for sanity and also don't want to duplicate insertion.
I'll add comment.

File: store-tests (hunk 1 above)
Comment (marusak) [reply store-tests]: Oh right, I just saw listing of states so I added it without reading what it is really about. Removed it.

**Comment by marusak** [comment]
Thanks, updated

**Review by martinpitt — APPROVED** [review]
:+1:

**Pull request merged**